src/metabase/middleware.clj

```diff
@@ -354,3 +361,82 @@
            (handler request))
          (catch Throwable e
            {:status 400, :body (.getMessage e)}))))
+
+
+;;; ------------------------------------------------------------ EXCEPTION HANDLING ------------------------------------------------------------
+
+
+(def ^:private ^:const streaming-response-keep-alive-interval-ms
+  "Interval between sending newline characters to keep Heroku from terminating
+   requests like queries that take a long time to complete."
+  (* 1 1000))
+
+;; Handle ring response maps that contain a core.async chan in the :body key:
+;;
+;; {:status 200
+;;  :body (async/chan)}
+;;
+;; and send each string sent to that queue back to the browser as it arrives
+;; this avoids output buffering in the default stream handling which was not sending
+;; any responses until ~5k characters where in the queue.
+(extend-protocol protocols/StreamableResponseBody
+  clojure.core.async.impl.channels.ManyToManyChannel
+  (write-body-to-stream [output-queue _ ^OutputStream output-stream]
+    (log/debug (u/format-color 'green "starting streaming request"))
+    (with-open [out (io/writer output-stream)]
+      (loop [chunk (async/<!! output-queue)]
+        (when-not (= chunk ::EOF)
+          (.write out (str chunk))
+          (try
+            (.flush out)
+            (catch org.eclipse.jetty.io.EofException e
+              (log/info (u/format-color 'yellow "connection closed, canceling request %s" (type e)))
+              (async/close! output-queue)
+              (throw e)))
+          (recur (async/<!! output-queue)))))))
+
+(def ^:private ^:const streaming-response-keep-alive-interval-ms
+  "Interval between sending whitespace bytes to keep Heroku from terminating
+   requests like queries that take a long time to complete."
```

Comment:
It would be helpful for other devs (your future self included) I think if you added a sentence on what the factors were in choosing this number. E.g. something like 

```
"Interval between sending whitespace bytes to keep Heroku from terminating
  requests like queries that take a long time to complete.
  This should be a good balance between blowing up the server by doing the check too often, or waiting 
  too long to return finished results."
```

(That's just a sample explanation, no idea why you actually picked it.)

This applies to the other constant as well